irobotclient/request_formatter.py

```diff
@@ -0,0 +1,84 @@
+"""
+Copyright (c) 2017 Genome Research Ltd.
+
+This program is free software: you can redistribute it and/or modify it
+under the terms of the GNU General Public License as published by the
+Free Software Foundation, either version 3 of the License, or (at your
+option) any later version.
+
+This program is distributed in the hope that it will be useful, but
+WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+Public License for more details.
+
+You should have received a copy of the GNU General Public License along
+with this program. If not, see <http://www.gnu.org/licenses/>.
+"""
+
+import os
+
+EXT_MAPPING = {
+    ".cram": (".crai",),
+    ".bam":  (".bai", ".pbi")
+}
+
+
+def get_header(auth_token: str) -> dict:
+    """
+    Set the request headers and return them as a dictionary.
+
+    :param auth_type:
+    :param auth_token:
+    :return:
+    """
+
+    headers = {}
+
+    # If an authorisation token has been supplied then it is likely Arvados authentication.
+    # Basic authorisation will be attempted after an unsuccessful request when the response specifies what
+    # authenitcation type it is expecting.
+    if auth_token is not None:
+        headers["authorization"] = f'Arvados {auth_token}'
+
+    return headers
+
+
+def get_url_request_path(irobout_url: str, input_file: str) -> str:
```

Comment:
It may seem like a silly comment, but typos like these will drive you crazy when it comes to debugging.